p256/src/arithmetic.rs

```diff
@@ -1,7 +1,16 @@
 //! A pure-Rust implementation of group operations on secp256r1.
 
+#[cfg(feature = "expose-arithmetic")]
+pub mod field;
+#[cfg(not(feature = "expose-arithmetic"))]
 mod field;
+#[cfg(feature = "expose-arithmetic")]
+pub mod scalar;
+#[cfg(not(feature = "expose-arithmetic"))]
 mod scalar;
+#[cfg(feature = "expose-arithmetic")]
+pub mod util;
+#[cfg(not(feature = "expose-arithmetic"))]
 mod util;
```

Comment:
It'd be nice to figure out a better solution for this sort of gating.

One option would be to make the arithmetic module `mod arithmetic` (no `pub`), and then re-export the types it contains at the toplevel when the `arithmetic` feature is enabled.

Since `doc_cfg` tags everything with the requisite cargo features, I don't think this would be confusing:

https://docs.rs/p256/0.3.0/p256/arithmetic/index.html

Reply:
I think the easiest solution will be to use the [`cfg_if`](https://docs.rs/cfg-if) crate.

Reply:
Yeah, that would work too.

For what it's worth I opened a PR with my suggested refactoring and I like it:

https://github.com/RustCrypto/elliptic-curves/pull/58

Reply:
Mmh, agreed. This is just a hack to allow use of the arithmetic in an experimental/hazmat manner outside the crate while core pieces are missing. I actually think the arithmetic should be public, but that's another topic - this PR is just to exemplify a possible ECDSA implementation.

Reply:
@nickray now that #58 is merged, you can refactor this to use something like the following in the toplevel of the crate:

```rust
#[cfg(feature = "expose-arithmetic"))]
#[cfg_attr(docsrs, doc_cfg(feature = "expose-arithmetic"))]
pub use arithmetic::{field, scalar, util};
```